scripts/github_bot/api/app.py

```diff
@@ -68,27 +93,65 @@ def _parse_pr_title(title):
+    check_call(['git', 'clone', 'https://github.com/{}'.format(ENV_REPO_NAME), working_dir])
+    check_call(['pip', 'install', '-e', 'scripts'], cwd=working_dir)
     check_call(['python', '-m', 'scripts.automation.release.run', '-c', component_name,
                 '-r', ENV_PYPI_REPO], cwd=working_dir)
     shutil.rmtree(working_dir, ignore_errors=True)
 
+def apply_release_label(issue_url):
+    payload = [RELEASE_LABEL]
+    r = requests.post('{}/labels'.format(issue_url), json=payload, auth=GITHUB_API_AUTH, headers=GITHUB_API_HEADERS)
+    return True if r.status_code in [200, 201] else False
+
+def comment_on_pr(comments_url, comment_body):
+    payload = {'body': comment_body}
+    r = requests.post(comments_url, json=payload, auth=GITHUB_API_AUTH, headers=GITHUB_API_HEADERS)
+    return True if r.status_code == 201 else False
+
+def create_github_release(component_name, component_version, released_pypi_url):
+    tag_name = '{}-{}'.format(component_name, component_version)
+    release_name = "{} {}".format(component_name, component_version)
+    payload = {'tag_name': tag_name, "target_commitish": "master", "name": release_name, "body": GITHUB_RELEASE_BODY_TMPL.format(released_pypi_url), "prerelease": True}
+    r = requests.post('https://api.github.com/repos/{}/releases'.format(ENV_REPO_NAME), json=payload, auth=GITHUB_API_AUTH, headers=GITHUB_API_HEADERS)
+    return True if r.status_code == 201 else False
+
 def _handle_pr_event(payload):
     pr_action = payload['action']
     pr = payload['pull_request']
     pr_merged = pr_action == 'closed' and pr['merged']
-    if not pr_merged:
-        return jsonify(msg="Ignoring event. PR not merged.")
     try:
         component_name, component_version = _parse_pr_title(pr['title'])
-        try:
-            create_release(component_name)
-            released_pypi_url = '{}/{}'.format(ENV_PYPI_REPO, component_name)
-            return jsonify(msg="Release of '{}' with version '{}' successful. View at {}".format(component_name, component_version, released_pypi_url))
-        except CalledProcessError as e:
-            print("Release of '{}' with version '{}' failed:".format(component_name, component_version))
-            print(e)
-            return jsonify(msg="Release of '{}' with version '{}' failed. View server logs for more info.".format(component_name, component_version)), 500
+        issue_url = pr['_links']['issue']['href']
+        comment_url = pr['_links']['comments']['href']
+        if pr_action == 'opened':
+            apply_release_label(issue_url)
+            comment_created = comment_on_pr(comment_url, OPENED_RELEASE_PR_COMMENT)
+            if comment_created:
+                return jsonify(msg="Commented on PR with release PR checklist.")
+            else:
+                return jsonify(msg="Attempted to comment on PR but API request did not succeed.")
+        elif not pr_merged:
+            return jsonify(msg="Ignoring event. PR not merged.")
```

Comment:
The server has to return some response (otherwise the request fails) so I opted to return a 200 response and have that message.